libs/community/langchain_community/document_loaders/parsers/pdf.py

```diff
@@ -119,16 +119,21 @@ def _extract_text_from_page(page: "PageObject") -> str:
                 )
 
         with blob.as_bytes_io() as pdf_file_obj:  # type: ignore[attr-defined]
-            pdf_reader = pypdf.PdfReader(pdf_file_obj, password=self.password)
-
-            yield from [
-                Document(
-                    page_content=_extract_text_from_page(page=page)
-                    + self._extract_images_from_page(page),
-                    metadata={"source": blob.source, "page": page_number},  # type: ignore[attr-defined]
-                )
-                for page_number, page in enumerate(pdf_reader.pages)
-            ]
+            try:
+                pdf_reader = pypdf.PdfReader(pdf_file_obj, password=self.password)
+            except pypdf.errors.PyPdfError as e:
+                raise e
+            try:
+                yield from [
+                    Document(
+                        page_content=_extract_text_from_page(page=page)
+                        + self._extract_images_from_page(page),
+                        metadata={"source": blob.source, "page": page_number},  # type: ignore[attr-defined]
+                    )
+                    for page_number, page in enumerate(pdf_reader.pages)
+                ]
+            except (pypdf.errors.PyPdfError, RecursionError) as e:
```

Comment:
Consider this https://github.com/py-pdf/pypdf/issues/2761. There are some other exceptions that can be thrown by pypdf library. Maybe you should catch any Exception here, or wait until issue is closed in pypdf.

Reply:
Sure I will do add other exceptions as well.